### PR TITLE
[EFR32] Fix OpenThread and Matter shell operation on EFRMG24 boards

### PR DESCRIPTION
**Pull request by selissia**
#### Problem
OpenThread and Matter shells do not work on EFRMG24 boards

#### Change overview
Use correct platform calls for EUSART initialization and IRQ handling. 

#### Testing
Tested various commands in the OpenThread and Matter shells on BRD4186A (MG24) and BRD4161A (MG12).

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking platform changes.

File: examples/platform/efr32/uart.cpp [hunk 1]
```diff
@@ -44,18 +44,27 @@ extern "C" {
 #define MIN(A, B) ((A) < (B) ? (A) : (B))
 #endif
 
+#ifdef EFR32MG24
+#define HELPER1(x) EUSART##x##_RX_IRQn
```
Comment (andy31415) [inline examples/platform/efr32/uart.cpp:48]: what is "HELPER" ? Why is this not "EUSART_IRQ_NO" or something else that is clearer in wording and more readable?

below then we do `#define HELPER2(x) HELPER1(x)` ... why?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/platform/efr32/uart.cpp (hunk 1 above)
Comment (andy31415) [reply examples/platform/efr32/uart.cpp:48]: Resolving because platform code... but personal opinion is that this is not good for maintenance.

Comment (selissia) [reply examples/platform/efr32/uart.cpp:48]: This code is copied from Silicon Labs' platform examples. 

> what is "HELPER" ? Why is this not "EUSART_IRQ_NO" or something else that is clearer in wording and more readable?

There's no good reason, that's just how it is. 

> below then we do #define HELPER2(x) HELPER1(x) ... why?

That's a limitation of gcc preprocessor. "If you want to stringify the result of expansion of a macro argument, you have to use two levels of macros." https://gcc.gnu.org/onlinedocs/gcc-4.8.5/cpp/Stringification.html

**Pull request merged**